packages/react-data-grid/src/masks/SelectionMask.js

```diff
@@ -1,43 +1,15 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import CellMask from './CellMask';
-import * as columnUtils from '../ColumnUtils';
-import zIndexes from 'common/constants/zIndexes';
-
-const isFrozenColumn = (columns, { idx }) => columnUtils.isFrozen(columnUtils.getColumn(columns, idx));
 
-const isScrollingHorizontallyWithoutCellChange = ({ scrollTop, prevScrollTop, scrollLeft, prevScrollLeft, selectedPosition, prevSelectedPosition }) => {
-  return scrollLeft !== prevScrollLeft && (scrollTop === prevScrollTop) && selectedPosition.idx === prevSelectedPosition.idx;
-};
-
-const getLeftPosition = (isFrozen, cellLeft, props) => {
-  if (isFrozen && !isScrollingHorizontallyWithoutCellChange(props)) {
-    return props.scrollLeft + cellLeft;
-  }
-  return cellLeft;
-};
-
-
-export const getCellMaskDimensions = (props) => {
-  const { selectedPosition, columns, getSelectedRowHeight, getSelectedRowTop } = props;
-  const column = columnUtils.getColumn(columns, selectedPosition.idx);
-  const height = getSelectedRowHeight(selectedPosition.rowIdx);
-  const top = getSelectedRowTop(selectedPosition.rowIdx);
-  const frozen = isFrozenColumn(columns, selectedPosition);
-  const zIndex = frozen ? zIndexes.FROZEN_CELL_MASK : zIndexes.CELL_MASK;
-  const left = getLeftPosition(frozen, column.left, props);
-  return { height, top, width: column.width, left, zIndex };
-};
+import CellMask from './CellMask';
+import { getSelectedDimensions } from '../utils/SelectedCellUtils';
```

Comment:
This component can be really simplified if we remove `getSelectedRowHeight` and `getSelectedRowTop`

Reply:
Agreed but I can see some users getting upset that their variable row height grids no longer work. We need to think of a better solution to handle dynamic row heights